src/BTVNanoCommissioning/utils/xs_scaler.py

```diff
@@ -1,20 +1,96 @@
-def read_xs(file):
-    import json
+import copy
+import hist
+from coffea import processor
+import os
+from BTVNanoCommissioning.helpers.xsection import xsection
 
-    f = open("../metadata/%s" % file)
-    data = json.load(f)
+
+def scale_xs(hist, lumi, events):
     xs_dict = {}
-    for obj in data:
+    for obj in xsection:
         xs_dict[obj["process_name"]] = float(obj["cross_section"])
-    return xs_dict
-
-
-def scale_xs(hist, lumi, events, xsfile="../metadata/xsection.json"):
-    xs_dict = read_xs(xsfile)
     scales = {}
     for key in events:
-        if type(key) != str or key == "Data":
+        if type(key) != str or key == "Data" or "Run" in key:
             continue
         scales[key] = xs_dict[key] * lumi / events[key]
     hist.scale(scales, axis="dataset")
     return hist
+
+
+def scale_xs_arr(events, lumi):
+    xs_dict = {}
+    for obj in xsection:
+        xs_dict[obj["process_name"]] = float(obj["cross_section"])
+    scales = {}
+    wei_array = {}
+    for key in events:
+        if type(key) != str or key == "Data" or "Run" in key:
+            continue
+        scales[key] = xs_dict[key] * lumi / events[key]
+
+        wei_array[key] = scales[key]
+    return wei_array
+
+
+def scaleSumW(accumulator, lumi, sumw, dyscale=1.0, xsfile="xsection.json"):
```

Comment:
compare with my comment on file xsection.py: apparently, you are using both the json and the python version of xscetion. Could you adjust the code such that only one of them will be needed?

Reply:
Indeed `xsfile` is not used in my current code, and I think it would be better to use only single xsection file to avoid confusion

Reply:
I agree, so please move to the python version of the section file (if I understood correctly, the python version is preferred over the json one)